libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/PredictiveCacheController.kt

```diff
@@ -119,6 +129,46 @@ class PredictiveCacheController @JvmOverloads constructor(
         mapListeners[map] = onStyleLoadedListener
     }
 
+    /**
+     * Create cache controllers for a map instance.
+     * Call when a new map instance is available.
+     *
+     * @param map an instance of [MapboxMap]
+     * @param styles a list of Mapbox style URIs to cache.
+     * If no styles are passed current map's style will be cached.
+     * Only styles hosted on Mapbox Services are supported.
+     * Only non-volatile styles will be cached.
+     */
+    @JvmOverloads
+    fun createStyleMapControllers(map: MapboxMap, styles: List<String> = emptyList()) {
+        val tileStore = map.getResourceOptions().tileStore
+        if (tileStore == null) {
+            handleError("TileStore instance not configured for the Map.")
+            return
+        }
+
+        val offlineManager = OfflineManager(map.getResourceOptions())
+
+        mapListeners[map]?.let {
+            removeMapControllers(map)
+        }
+
```

Comment:
Should we also create the controllers with the new `TileDescriptor` implementation or is it fine to only do that when the style is updated? That's what we were doing 👀 

https://github.com/mapbox/mapbox-navigation-android/blob/0059f31cbf376507720d1e54009e694e47624684/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/PredictiveCacheController.kt#L106-L113

Reply:
can we keep it simple and wait for style loaded? can we miss anything in this case?

Reply:
Do you know @kkaefer @LukasPaczos?

Reply:
Following up in https://github.com/mapbox/mapbox-navigation-android/pull/5068#discussion_r742967587.

Reply:
@LukasPaczos and I briefly discussed it internally and it seems we shouldn't wait for the style to load to create the controllers https://github.com/mapbox/mapbox-navigation-android/pull/5068#discussion_r742967587

Reply:
done